lib/index.js

```diff
@@ -372,6 +394,28 @@ function opt (option) {
   }
 };
 
+/* chan and rmchan sockopt methods. only relevant for subscription sockets */
+Socket.prototype.chan       = function(){
```

Comment:
One of the issues I have with the current API is that during socket construction, you can pass either a comma separated string, or an array.  After construction, you can call the chan method with only an array.

Since you might want to filter on chans that contain a comma, splitting on the comma in the constructor seems kind of brittle.  Rather than know about that edge case, why don't we eliminate it and require library consumers to have to always pass an array of strings?